miden/examples/nprime/nprime.masm

```diff
@@ -0,0 +1,95 @@
+# push the current prime+count tuple to the memory
+proc.push
+    # define the index of the prime to be added
+    dup.1
+
+    # push the current candidate to the primes list
+    mem_store
+end
+
+# push the next prime to the stack with the primes count
+proc.next
+    # initial state
+    # [last prime, primes count]
+
+    # push candidate & not found flag
+    # [not found, candidate, last prime, primes count]
+    dup
+    push.1
+
+    while.true
+        # increment the candidate
+        # [accumulator, candidate, last prime, primes count]
+        add.1
+        push.0
+
+        # push counters
+        # [i < n, i, n, accumulator, candidate, last prime, primes count]
+        dup.3
+        push.0
+        push.1
+
+        while.true
+            # fetch the candidate
+            # [candidate, i, n, accumulator, candidate, last prime, primes count]
+            dup.3
+
+            # fetch the current prime
+            # [prime, candidate, i, n, accumulator, candidate, last prime, primes count]
+            dup.1
+            add.1
+            mem_load
+
+            # calculate the mod prime
+            # this is a naive scan for the sake of simplicity. normally we should check up to sqrt of prime
+            # [remainder, i, n, accumulator, candidate, last prime, primes count]
+            u32checked_mod
+
+            # check if is divisor and update the accumulator
+            # [i, n, accumulator, candidate, last prime, primes count]
+            u32checked_eq.0
+            dup.3
+            add
+            swap.3
+            drop
+
+            # increment i up to n
+            add.1
+            dup
+            dup.2
+            u32checked_lt
+        end
+
+        # loop finished. if accumulator > 0, then the number has at least one divisor
+        # [accumulator, candidate, last prime, primes count]
+        drop
+        drop
+        u32checked_neq.0
+    end
+
+    # increment the prime counter
+    movup.2
+    add.1
+
+    # return the calculated prime
+    swap
+    exec.push
+end
+
+begin
+    # push the first prime
+    exec.push
+
+    # init the loop
+    push.1
+    while.true
+        # push the next prime + tuple count to the stack, and update the memory
+        exec.next
+
+        # check if nth prime
+        # in this example, we are arbitrarily checking the 24th prime - it can be any number greater than 1.
+        dup.1
+        push.24
+        u32checked_neq
```

Comment:
That is definitely possible. I just kept the primes count in the stack as a convenience. We need to keep track of that number anyway because we need to know how many times we iterate in the memory.

I'm not using the position `0` of the memory so I could alternatively keep the number there too. Do you have a preference of one over the other?

Reply:
Oh I meant something much simpler than this - I was just thinking you could `swap drop` after the loop so that it's not part of the output. But again - this is completely a matter of your preference for what you want the output to look like. I think the way you've implemented it with the count in the stack makes sense, and saving it in memory would probably be less efficient